src/utilities/paramValidation.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { describe, expect, test } from 'vitest'
+import { Route } from '@/types'
+import { flattenRoutes, routeParamsAreValid, path } from '@/utilities'
+
+const component = { template: '<div>This is component</div>' }
+
+describe('routeParamsAreValid', () => {
+  test('given route WITHOUT params, always return true', () => {
+    const route: Route = {
+      name: 'no-params',
+      path: '/no-params',
+      component,
+    }
+    const [flat] = flattenRoutes([route])
+
+    const response = routeParamsAreValid('/no-params', flat)
+
+    expect(response).toBe(true)
+  })
+
+  test('given route with simple string param and value present, returns true', () => {
+    const route: Route = {
+      name: 'simple-params',
+      path: '/simple/:simple',
+      component,
+    }
+    const [flat] = flattenRoutes([route])
+
+    const response = routeParamsAreValid('/simple/ABC', flat)
+
+    expect(response).toBe(true)
+  })
+
+  test('given route with simple string param WITHOUT value present, returns false', () => {
+    const route: Route = {
+      name: 'simple-params',
+      path: '/simple/:simple',
+      component,
+    }
+    const [flat] = flattenRoutes([route])
+
+    const response = routeParamsAreValid('/simple', flat)
+
+    expect(response).toBe(false)
+  })
+
+  test('given route with OPTIONAL string param WITHOUT value present, returns true', () => {
+    const route: Route = {
+      name: 'simple-params',
+      path: '/simple/:?simple',
+      component,
+    }
+    const [flat] = flattenRoutes([route])
+
+    const response = routeParamsAreValid('/simple', flat)
```

Comment:
I think this test is incorrect. It shouldn't match `/simple` because it's missing the `/` at the end right?

Reply:
100%. This test was failing too and I was just about to dig into why. Thanks @pleek91 👏